Fix call to liquidprompt function

DarrenBishop opened the pull request:
Missing leading underscore in `_lp_escape`

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Without this - presumably - the code fails silently, never presenting the branch name in the PS1 prompt.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
Forgive me if there are tests for themes, but I could not find any; I personally tested this by cd'ing into and out of Git clone directories with and without the fix.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

davidpfarrell commented:
Relevant information from [Version 2.0 Upgrade Notes](https://github.com/nojhan/liquidprompt/blob/master/docs/upgrading/v2.0.rst#lp-escape) (dated Oct 6): 

_lp_escape()
------------

Renamed to `__lp_escape`.
Return changed from stdout to ``$ret``

Replace assignment statements like:

    output="$(_lp_escape "$input")"


with:

    local ret
    __lp_escape "$input"
    output=$ret

davidpfarrell commented:
Best I can tell this meets the requirements for the recent 2.0 version, but if we publish it as-is, isn't it going to break the existing installations?

Not sure what the best way to deal with it is?

DarrenBishop commented:
@davidpfarrell I am on the case

DarrenBishop commented:
@davidpfarrell 
> Not sure what the best way to deal with it is?

It got a bit hairy at one point, but what I have done is:

Introduced some auxiliary functions 
  - `_lp_legacy` to detect if <`v2.y.z` is in use by presence of the old `_lp_escape`
  - `__lp_escape`, but only if in _legacy_ mode; it routes to `__lp_escape` and uses the `ret` return variable

These functions are unset before the sourcing of Liquidprompt, to provide a clean slate if reloading the shell etc.

Updated `_lp_git_branch` to use the `lp_vcs_branch` return variable, but also print to stdout and (importantly) `return 0` when in _legacy_ mode.

I think a good practice going forward would be to pin the version of Liquidpromt to a tagged version, maybe via some env var, and print a warning/disclaimer if it's ever changed from the default.
